spec/components/projects/index_sub_header_component_spec.rb

```diff
@@ -0,0 +1,196 @@
+    end
+
+    with_request_url "/projects" do
+      render_inline(described_class.new(current_user:, query:))
+    end
+  end
+
+  describe "add buttons" do
+    workspace_types = %i[portfolio program project] # rubocop:disable RSpec/LeakyLocalVariable
+
+    let(:add_label) { I18n.t(:button_add) }
+    let(:portfolio_label) { I18n.t(:label_portfolio) }
+    let(:program_label) { I18n.t(:label_program) }
+    let(:project_label) { I18n.t(:label_project) }
+
+    subject { page }
+
+    shared_examples "renders buttons" do |expected_types:, upsell: []|
+      workspace_types.each do |workspace_type|
+        if expected_types.include?(workspace_type)
+          it "renders #{workspace_type} add button" do
+            label = send("#{workspace_type}_label")
+            href = send("new_#{workspace_type}_path")
+
+            expect(subject).to have_link(label, href:)
+
+            link = subject.find_link(label)
+
+            if upsell.include?(workspace_type)
+              expect(link).to have_octicon "op-enterprise-addons"
+            else
+              expect(link).to have_no_octicon "op-enterprise-addons"
+            end
+          end
+        else
+          it "doesn't render #{workspace_type} add button" do
+            expect(subject).to have_no_link send("#{workspace_type}_label")
+          end
+        end
+      end
+    end
+
+    shared_examples "renders no buttons" do
+      it { is_expected.to have_no_button add_label }
+
+      include_examples "renders buttons", expected_types: []
+    end
+
+    shared_examples "renders add button directly" do |expected_type, upsell: false|
+      it { is_expected.to have_no_button add_label }
+
+      include_examples "renders buttons", expected_types: [expected_type], upsell: upsell ? [expected_type] : []
+    end
+
+    shared_examples "renders add buttons in a pulldown" do |expected_types, upsell: []|
+      it { is_expected.to have_button add_label }
+
+      context "within pulldown" do
+        subject { page.find(:button, add_label).find(:xpath, "..//action-list") }
+
+        include_examples "renders buttons", expected_types:, upsell:
+      end
+    end
+
+    context "when user has no permissions" do
+      let(:global_permissions) { [] }
+
+      include_examples "renders no buttons"
+    end
+
+    context "when user has add_portfolios permission" do
+      let(:global_permissions) { %i[add_portfolios] }
+
+      context "without feature flag", with_flag: { portfolio_models: false } do
+        include_examples "renders no buttons"
+      end
+
+      context "with feature flag", with_flag: { portfolio_models: true } do
+        context "without enterprise feature enabled", with_ee: [] do
+          include_examples "renders add button directly", :portfolio, upsell: true
+        end
+
+        context "with enterprise feature enabled", with_ee: :portfolio_management do
+          include_examples "renders add button directly", :portfolio
```

Comment:
Just a nitpick, adding the `upsell: false` explicitly, will make the spec more readable.
```suggestion
          include_examples "renders add button directly", :portfolio, upsell: false
```

Reply:
Good point, made the argument required and changed all calls appropriately